### PR TITLE
chore: port to numpy 2.0

### PR DESCRIPTION
**Pull request by ianna**
fixes issue #56

### CODE REVIEW

File: tests/test_spec_elementwise_functions.py [hunk 1]
```diff
@@ -14,13 +14,28 @@
...
+#     with warnings.catch_warnings():
+#         warnings.simplefilter("ignore")
+#         import array_api_strict as xp  # type: ignore[import-not-found]
+
+#         has_complex_dtype = np.dtype("complex128") in xp._dtypes._all_dtypes
+# else:
+#     xp = np
 devices = ["cpu"]
+try:
+    import numpy.array_api as xp
```
Comment (ianna) [inline tests/test_spec_elementwise_functions.py:34]:
```suggestion
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
         import numpy.array_api as xp
```

**Review by ianna — COMMENTED** [review]
@ohrechykha - the tests with Numpy version 1 are failing because `numpy.array_api` is experimental. Please, add the ignore statement as above.

**Comment by ianna** [comment]
@ohrechykha - looks good! There are two more issues left. You can check them by clicking on "Details". One is the lower bound testing (issue #61). It should be fixed in a separate PR, I think. And another is related to the differences in Numpy versions (`asarray`) that should be fixed in this PR.

**Comment by ohrechykha** [comment]
I'm unsure what's wrong: the idea was that defining _wrapper in floor and ceil functions is unnecessary (because I defined it in ceil but forgot to add it to floor and nothing happened). After seeing the failures I've concluded this was wrong and returned them. The ubuntu tests still fail with the same errors, while on Windows I haven't gotten any errors regardless of whether there's a wrapper function in ceil and floor, meaning it could likely be removed.

**Comment by ianna** [comment]
> I'm unsure what's wrong: the idea was that defining _wrapper in floor and ceil functions is unnecessary (because I defined it in ceil but forgot to add it to floor and nothing happened). After seeing the failures I've concluded this was wrong and returned them. The ubuntu tests still fail with the same errors, while on Windows I haven't gotten any errors regardless of whether there's a wrapper function in ceil and floor, meaning it could likely be removed.

The tests in the CI pass :-)

**Comment by ohrechykha** [comment]
Oh well. I've made it work for numpy 1.22.3, while git is testing it on 1.22.0...

**Comment by hmaarrfk** [comment]
is numpy 1.22 a version you intend on supporting? Numpy recommends not supporting lower than 1.24 in june of 2024.

https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table

Just trying to help make these chores easier!

**Comment by jpivarski** [comment]
NumPy 1.22 was chosen as a minimum for Awkward Array, based on functionality. (Something was added in NumPy 1.22 that Awkward requires; no tighter requirements were needed after that.) We can get on the train of always-moving dependency windows, but we just haven't yet.

If ragged needs something that NumPy 1.22 doesn't provide—that is, if it works in a later version of NumPy but not 1.22—then let's just increase the minimum NumPy version as needed, especially since it agrees with NumPy's own schedule of support.

In principle, we _should_ be moving our version constraints according to the SPEC 0 schedule; we just haven't started that yet.

**Comment by ianna** [comment]
@jpivarski and @ohrechykha - it looks like Numpy 2.1.0 went back to return the same `dtype` as its elements `dtype`.

**Comment by ianna** [comment]
@jpivarski - Please, review when you have time. Thanks!

**Pull request merged**